src/lib/modules/Patron/PatronBulkExtendLoans/actions.js

```diff
@@ -0,0 +1,59 @@
+import { loanApi } from '@api/loans';
+import { searchReady } from '@api/utils';
+import {
+  sendErrorNotification,
+  sendSuccessNotification,
+} from '@components/Notifications';
+import {
+  fetchPatronCurrentLoans,
+  IS_LOADING as currentLoansIsLoading,
+} from '../PatronCurrentLoans/actions';
+import _isEmpty from 'lodash/isEmpty';
+
+export const IS_LOADING = 'bulkLoanExtend/IS_LOADING';
+export const SUCCESS = 'bulkLoanExtend/SUCCESS';
+export const HAS_ERROR = 'bulkLoanExtend/HAS_ERROR';
+
+export const bulkLoanExtension = (patronPid) => {
+  return async (dispatch) => {
+    dispatch({
+      type: IS_LOADING,
+    });
+    try {
+      const response = await loanApi.bulkExtendLoans(patronPid);
+
+      const notExtendedMessage = ` ${response.data.not_extended_loans.length} 
```

Comment:
I see the need for the extra space, but I have the feeling it will be removed by someone else (probably me) at some point thinking that it is a mistake :)
I propose a refactor to, hopefully, improve readability and decrease the complexity of folliwing the code with the various `if`.

```
const extendedLoansMsg = response.data.extended_loans ?`${response.data.extended_loans.length}  loan(s) have been successfully extended.` : '';
const notExtendedLoansMsg = response.data.not_extended_loans ? `${response.data.not_extended_loans.length} loan(s) could not be extended. Consult the ongoing loans section or contact the library for more details.` : '';

// no need to use isEmpty for lists
if (response.data.extended_loans) {
    const msg = `{extendedLoansMsg} {notExtendedLoansMsg}`; // the extra space in the middle should not hurt when notExtendedLoansMsg is empty string
  dispatch...
} else {
  // the extra msg "There was no loans eligible for an extension." is probaly not needed, because basically all loans cannot be extended, so why not simply showing the msg above?
  dispatch(sendSuccessNotification('No loans extended', notExtendedLoansMsg));
}

```